### ISSUE TITLE
azurerm_synapse_workspace identity issue

### ISSUE THREAD

**Issue by farwind** [hashicorp/terraform-provider-azurerm#25761]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.7

### AzureRM Provider Version

3.100.0

### Affected Resource(s)/Data Source(s)

azurerm_synapse_workspace

### Terraform Configuration Files

```hcl
minimal build of synapse workspace based on terraform registry docs.
without identity block since that is marked as optional.
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

synapse workspace created

### Actual Behaviour

 Error: creating Workspace: (Name "synw-cancom-ci-synapse-ytb9-we-test" / Resource Group "rg-cancom-ci-synapse-ytb9-we-test"): synapse.WorkspacesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="ValidationFailed" Message="Workspace request validation failed, check error details for more information" Details=[{"code":"SystemAssignedManagedIdentityNotSpecified","message":"Workspace creation request should have an identity section with Type = SystemAssigned"}]
│ 
│   with module.defaults.azurerm_synapse_workspace.this[0],
│   on ../../main.tf line 54, in resource "azurerm_synapse_workspace" "this":
│   54: resource "azurerm_synapse_workspace" "this" {

### Steps to Reproduce

create resource group, create storage account, in that create data_lake_gen2_filesystem, take all required variables based on the terraform registry and try to spin up synapse workspace.

### Important Factoids

no

**Comment by farwind** [comment]
https://registry.terraform.io/providers/hashicorp/azurerm/3.100.0/docs/resources/synapse_workspace
documentation required parameters do not match criteria for synapse_workspace to be sucesfully built.

**Comment by Chambras** [comment]
@farwind you are right. It failed when using only the required values in the documentation. I added 
```tf
  identity {
    type = "SystemAssigned"
  }
```
and it worked. Do not know if it is documentation issue and it needs to say identity block is `required` or if there is a bug and if the block is not set, it needs to default to `SystemAssigned`
